internal/operator/statefulset/stsmatcher.go

```diff
@@ -0,0 +1,587 @@
+		return matchVolMount(common.ProviderBackupSecretVolumeName, common.GCSBackupSecretVolumeMountPath)
+	case utils.S3, utils.ABS, utils.OSS, utils.Swift, utils.OCS:
+		return matchVolMount(common.ProviderBackupSecretVolumeName, common.NonGCSProviderBackupSecretVolumeMountPath)
+	}
+	return nil
+}
+
+func (s StatefulSetMatcher) matchEtcdContainerEnvVars() gomegatypes.GomegaMatcher {
+	if s.useEtcdWrapper {
+		return BeEmpty()
+	}
+	scheme := utils.IfConditionOr[string](s.etcd.Spec.Backup.TLS != nil, "https", "http")
+	return ConsistOf(
+		MatchFields(IgnoreExtras, Fields{
+			"Name":  Equal("ENABLE_TLS"),
+			"Value": Equal(strconv.FormatBool(s.etcd.Spec.Backup.TLS != nil)),
+		}),
+		MatchFields(IgnoreExtras, Fields{
+			"Name":  Equal("BACKUP_ENDPOINT"),
+			"Value": Equal(fmt.Sprintf("%s://%s-local:%d", scheme, s.etcd.Name, s.backupPort)),
+		}),
+	)
+}
+
+func (s StatefulSetMatcher) matchEtcdPodSecurityContext() gomegatypes.GomegaMatcher {
+	if !s.useEtcdWrapper {
+		Equal(nil)
+	}
+	return PointTo(MatchFields(IgnoreExtras|IgnoreMissing, Fields{
+		"RunAsGroup":   PointTo(Equal(int64(65532))),
+		"RunAsNonRoot": PointTo(Equal(true)),
+		"RunAsUser":    PointTo(Equal(int64(65532))),
+		"FSGroup":      PointTo(Equal(int64(65532))),
+	}))
+}
+
+func (s StatefulSetMatcher) matchPodVolumes() gomegatypes.GomegaMatcher {
+	volMatchers := make([]gomegatypes.GomegaMatcher, 0, 7)
+	etcdConfigFileVolMountMatcher := MatchFields(IgnoreExtras, Fields{
+		"Name": Equal(common.EtcdConfigVolumeName),
+		"VolumeSource": MatchFields(IgnoreExtras|IgnoreMissing, Fields{
+			"ConfigMap": PointTo(MatchFields(IgnoreExtras, Fields{
+				"LocalObjectReference": MatchFields(IgnoreExtras, Fields{
+					"Name": Equal(s.etcd.GetConfigMapName()),
+				}),
+				"Items": HaveExactElements(MatchFields(IgnoreExtras|IgnoreMissing, Fields{
+					"Key":  Equal(etcdConfigFileName),
+					"Path": Equal(etcdConfigFileName),
+				})),
+				"DefaultMode": PointTo(Equal(int32(0640))),
+			})),
+		}),
+	})
+	volMatchers = append(volMatchers, etcdConfigFileVolMountMatcher)
+	secretVolMatchers := s.getPodSecurityVolumeMatchers()
+	if len(secretVolMatchers) > 0 {
+		volMatchers = append(volMatchers, secretVolMatchers...)
+	}
+	if s.etcd.IsBackupStoreEnabled() {
+		backupVolMatcher := s.getBackupVolumeMatcher()
+		if backupVolMatcher != nil {
+			volMatchers = append(volMatchers, backupVolMatcher)
+		}
+	}
+	return ConsistOf(volMatchers)
+}
+
+func (s StatefulSetMatcher) getPodSecurityVolumeMatchers() []gomegatypes.GomegaMatcher {
+	volMatchers := make([]gomegatypes.GomegaMatcher, 0, 5)
+	if s.etcd.Spec.Etcd.ClientUrlTLS != nil {
+		volMatchers = append(volMatchers, MatchFields(IgnoreExtras, Fields{
+			"Name": Equal(common.EtcdCAVolumeName),
+			"VolumeSource": MatchFields(IgnoreExtras, Fields{
+				"Secret": PointTo(MatchFields(IgnoreExtras, Fields{
+					"SecretName":  Equal(s.etcd.Spec.Etcd.ClientUrlTLS.TLSCASecretRef.Name),
+					"DefaultMode": PointTo(Equal(int32(0640))),
```

Comment:
Could this be reflected for all references in this file?
https://github.com/gardener/etcd-druid/pull/777#discussion_r1592016707